qa-tests-backend/src/test/groovy/NetworkFlowTest.groovy

```diff
@@ -689,6 +688,42 @@ class NetworkFlowTest extends BaseSpecification {
         Services.waitForSRDeletion(delete)
 
         when:
+        "Network graph contains all expected targets"
+        NetworkGraph currentGraph = null
+
+        // This is a map of "deploymentName" -> expectedNumberOfEdges.
+        // It allows us to know when the network graph has reached a desired state and the proper assertions can start.
+        def targets = [(TCPCONNECTIONTARGET):3, (UDPCONNECTIONTARGET):1]
+        targets.each { deploymentName, expectedEdges ->
+            // Make sure that we have enough time to populate the graph: 30s for sensor + some extra.
+            // On the first run, this test benefits from the state left over by previous tests.
+            // However, if the first run fails, the env is recreated and the test is run immediately afterwards.
+            // The system requires time for the changes to propagate to sensor, thus we have an additional retry here.
+            log.info "Verifying graph edges of ${deploymentName}"
+            Helpers.withRetry(6, 20) { retry ->
+                // The graph needs to be re-retrieved on each retry
+                currentGraph = NetworkGraphService.getNetworkGraph()
+
+                def index = currentGraph.nodesList.findIndexOf { node -> node.deploymentName == deploymentName }
+                List<NetworkNode> outNodes = currentGraph.nodesList.findAll { node ->
+                    node.outEdgesMap.containsKey(index)
+                }
+                def allowAllIngress = deployments.find { it.name == deploymentName }?.createLoadBalancer ||
+                    currentGraph.nodesList.find { it.entity.type == Type.INTERNET }.outEdgesMap.containsKey(index)
+                if (allowAllIngress) {
+                    log.info "${deploymentName} has LB/External incoming traffic - ensure All Ingress allowed"
```

Comment:
[nit] I think this can be removed. It's already logged later in the `then` block.

Reply:
Oh, sure this is a duplication, but it helped me in debugging when I wondered why the assertions are not being triggered. I would like to keep it if that would be fine.

Reply:
Sure! No problem.